### ISSUE TITLE
ByteBufferBasicBytez lacks copyTo() implementation

### ISSUE THREAD

**Issue by abyrd** [RuedigerMoeller/fast-serialization#174]
ByteBufferBasicBytez lacks an implementation of the copyTo() method. This method is necessary for expanding buffers during serialization (see org.nustaq.serialization.coders.FSTBytezEncoder::ensureFree) and it is necessary for deserialization (e.g. reading class names out of the serialized data). 

The following implementation from MallocBytez seems to work fine:
```Java
@Override
    public void copyTo(BasicBytez other, long otherByteIndex, long myByteIndex, long lenBytes) {
        if ( other instanceof HeapBytez) {
            HeapBytez hp = (HeapBytez) other;
            unsafe.copyMemory(null,baseAdress+myByteIndex, hp.getBase(), hp.getOff()+otherByteIndex,lenBytes);
        } else {
            for ( long i = 0; i < lenBytes; i++ ) {
                other.put(otherByteIndex+i,get(myByteIndex+i));
            }
        }
    }
```

**Comment by RuedigerMoeller** [comment]
thanks a lot :)

**Comment by RuedigerMoeller** [comment]
fixed with 2.49
